python/Makefile

```diff
@@ -26,14 +26,21 @@ lint:
 	poetry run pre-commit run --all-files
 
 test:
-	poetry run coverage run --source=pyiceberg/ -m pytest tests/ -m "not s3" ${PYTEST_ARGS}
+	poetry run coverage run --source=pyiceberg/ -m pytest tests/ -m "not s3 and not adlfs" ${PYTEST_ARGS}
```

Comment:
The "not X and not Y" does not really scale well, could we use another scheme to enable/disable tests?  Comment also applies further down to `test-s3` and `test-adlfs`.

Reply:
We're open to better ideas!

Reply:
I've been looking at this in https://github.com/apache/iceberg/pull/6398/ as well. I've added:

```python
def pytest_collection_modifyitems(items, config):
    for item in items:
        if not any(item.iter_markers()):
            item.add_marker("unmarked")
```

Which will automatically add the mark `unmarked` to each of the tests that doesn't have a marked, creating their own group. WDYT of that?

Reply:
Yeah, that sounds good.  I'll leave my code as-is and if my PR goes in before yours, you'll have to modify this line as a conflict resolution.